### PR TITLE
[libc++][chrono] Implements duration Rep constraints.

### PR DESCRIPTION
**Pull request by mordante**
Applies LWG3050 to the constraints of operator*, operator/, and operator%. The changes to the constructor were done in
https://reviews.llvm.org/D118902, but that patch did not identify the related LWG-issue, and only adjusted the constructor to the wording in the Standard.

Implements:
- LWG 3050: Conversion specification problem in chrono::duration constructor

### CODE REVIEW

File: libcxx/test/std/time/rep.h [hunk 1]
```diff
@@ -29,6 +30,28 @@ class Rep
 
 struct NotARep {};
 
+#if TEST_STD_VER >= 11
+// Several duration operators take a Rep parameter. Before LWG3050 this
+// parameter was constrained to be convertible from a non-const object,
+// but the code always uses a const object. So the function was SFINEA's
+// away for this type. LWG3050 fixes the constrain to use a const
```
Comment (h-vetinari) [inline libcxx/test/std/time/rep.h:37]:
```suggestion
// but the code always uses a const object. So the function was SFINAE'd
// away for this type. LWG3050 fixes the constraint to use a const
```

**Review by ldionne — APPROVED** [review]
Thanks for the catch!

**Comment by h-vetinari** [comment]
While I'm flattered, I don't think a typo fix is worth a `Co-Authored-By:` line... 😅 
For me personally, feel free to drop that (auto-generated) line for trivial input going forward.

**Comment by mordante** [comment]
That is indeed automatically added when accepting a suggestion.

**Comment by h-vetinari** [comment]
I know, but at least it's possible to edit the commit message of the squash merge. ;)

**Pull request merged**